10_rest/runnerdave/pkg/puppy/rest.go

```diff
@@ -0,0 +1,98 @@
+package puppy
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	chi "github.com/go-chi/chi"
+)
+
+// RestStorer store for puppies
+type RestStorer struct {
```

Comment:
RestStorer does not seem like the right term.
1. `er` suffix usually indicates interface not implementation
2. Rest is to do with http request response not with storing. it might use a storer to persist/retrieve data, but its not its core purpose. Hope I make sense.

I usually call this type of struct `RestServer` or similar and make it implement the http.Handler interface so I can run it as `http.ListenAndServe(port, restServer)` - not 100% sure if that's great either....

Reply:
fixed